src/main/java/net/dv8tion/jda/internal/entities/InteractionEntityBuilder.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.entities.channel.concrete.*;
+import net.dv8tion.jda.api.entities.channel.middleman.GuildChannel;
+import net.dv8tion.jda.api.interactions.DiscordLocale;
+import net.dv8tion.jda.api.utils.data.DataArray;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.JDAImpl;
+import net.dv8tion.jda.internal.entities.channel.concrete.PrivateChannelImpl;
+import net.dv8tion.jda.internal.entities.channel.concrete.detached.*;
+import net.dv8tion.jda.internal.entities.channel.mixin.attribute.IInteractionPermissionMixin;
+import net.dv8tion.jda.internal.entities.channel.mixin.concrete.PrivateChannelMixin;
+import net.dv8tion.jda.internal.entities.detached.DetachedGuildImpl;
+import net.dv8tion.jda.internal.entities.detached.DetachedMemberImpl;
+import net.dv8tion.jda.internal.entities.detached.DetachedRoleImpl;
+import net.dv8tion.jda.internal.interactions.ChannelInteractionPermissions;
+import net.dv8tion.jda.internal.interactions.MemberInteractionPermissions;
+import net.dv8tion.jda.internal.utils.JDALogger;
+import org.slf4j.Logger;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class InteractionEntityBuilder extends AbstractEntityBuilder
+{
+    private static final Logger LOG = JDALogger.getLog(InteractionEntityBuilder.class);
+
+    private final EntityBuilder entityBuilder = api.getEntityBuilder();
```

Comment:
Does this compile?
It feels like `api` here shouldn't exist.

I think it'd make more sense to put the setter for this in the constructor, no?